src/main/java/uk/ac/ebi/eva/commons/core/models/VariantSourceEntryWithSamples.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.commons.core.models;
+
+import uk.ac.ebi.eva.commons.mongodb.entity.subdocuments.VariantSourceEntryMongo;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class VariantSourceEntryWithSamples extends AbstractVariantSourceEntry implements IVariantSourceEntry {
```

Comment:
Because we have two completely different structures that diverge at VariantSourceEntries and Variants This is a code smell that we need to change... but we can't change it until we do a new version of the rest API.